### ISSUE TITLE
withDevCycleProvider not forwarding ref

### ISSUE THREAD

**Issue by jeinhorn787** [DevCycleHQ/js-sdks#861]
Hello!  I am currently working with DVC's `@devcycle/devcycle-react-sdk` package and am using the `withDevCycleProvider` HOC. The component I am passing to the HOC requires a React `ref` be forwarded and so, that component is wrapped in [`forwardRef`](https://react.dev/reference/react/forwardRef) from React.  `withDevCycleProvider` does not return `WrappedComponent` wrapped in `forwardRef`.  This is causing the `ref` to be lost.
```
const Component = forwardRef((props, ref) => { ....... });

export default withDevCycleProvider(configObject)(Component);  <-- returns a WrappedComponent with no forwarded ref
```
**Environment**
`@devcycle/devcycle-react-sdk`: ^1.11.5
`node`: 18.18.0
`npm`: 9.8.1

**Comment by ajwootto** [comment]
Hey @jeinhorn787 thanks for the report, we'll take a look at fixing this.

In the meantime you can work around this by using the `DevCycleProvider` directly (the HOC just adds that component around your original component)

For example

```tsx

export const MyApp = () => {
    const ref = useRef()
    return (
        <DevCycleProvider config={{ sdkKey: 'YOUR_SDK_KEY', options: {} }}>
            <MyInnerComponent ref={ref} />
        </DevCycleProvider>
    )
}
```

**Comment by jonathannorris** [comment]
Also note that the [@devcycle/devcycle-react-sdk](https://www.npmjs.com/package/@devcycle/devcycle-react-sdk) repo has been deprecated for: [@devcycle/react-client-sdk](https://www.npmjs.com/package/@devcycle/react-client-sdk).

**Comment by jeinhorn787** [comment]
> Also note that the [@devcycle/devcycle-react-sdk](https://www.npmjs.com/package/@devcycle/devcycle-react-sdk) repo has been deprecated for: [@devcycle/react-client-sdk](https://www.npmjs.com/package/@devcycle/react-client-sdk).

Sure thing! Are there any known breaking issues going from our current version to the latest package?

**Comment by ajwootto** [comment]
@jeinhorn787 We've released version `1.23.1` of the SDK with a fix for the forwardRef behaviour. Let us know if you get a chance to try it!

Also, the two packages mentioned above should not have any breaking changes between each other, the `@devcycle/react-client-sdk` package is just a rename of the "deprecated" one.
